src/leveled_sst.erl

```diff
@@ -2161,115 +2171,102 @@ lookup_slots(StartKey, EndKey, Tree, FilterFun) ->
 
 -spec accumulate_positions(
     list(leveled_codec:ledger_kv()),
-    {
-        binary(),
-        non_neg_integer(),
-        list(leveled_codec:segment_hash()),
-        leveled_codec:last_moddate()
-    }
+    {non_neg_integer(), position_acc()}
 ) ->
-    {
-        binary(),
-        non_neg_integer(),
-        list(leveled_codec:segment_hash()),
-        non_neg_integer()
-    }.
+    position_acc().
 %% @doc
 %% Fold function use to accumulate the position information needed to
 %% populate the summary of the slot
-accumulate_positions(
-    [], {PosBin, NoHashCount, HashAcc, LMDAcc}
-) when is_integer(LMDAcc) ->
-    {PosBin, NoHashCount, HashAcc, LMDAcc};
-accumulate_positions([{K, V} | T], {PosBin, NoHashCount, HashAcc, LMDAcc}) ->
-    {_SQN, H1, LMD} = leveled_codec:strip_to_indexdetails({K, V}),
+accumulate_positions([], {NHC, {PosBin, HashAcc, LMDAcc}}) when
```

Comment:
Does this check for LMDAcc to be an integer have to be here?
It seems rather odd. The only place where you check for integer is here. The type specification already claims it is a `non_neg_integer`. So what are you protecting against here?

It would read simpler if it would read:
```
accumulate_positions([], {NHC, PosAcc}) ->
   finalise_posbin(NHC, PosAcc);
```